WKData/Sources/WKData/Data Controllers/WKGrowthTasks/WKGrowthTasksDataController.swift

```diff
@@ -0,0 +1,157 @@
+import Foundation
+
+public final class WKGrowthTasksDataController {
+
+    var service = WKDataEnvironment.current.mediaWikiService
+    let project: WKProject
+
+    public init (project: WKProject) {
+        self.project = project
+    }
+
+    // MARK: GET Methods
+
+    // TODO: to get more tasks, send used IDS to `ggtexcludepageids` parameter
+    // TODO: remove debug prints
+    public func getGrowthAPITask(task: WKGrowthTaskType = .imageRecommendation, completion: @escaping (Result<[WKGrowthTask.Page], Error>) -> Void) {
+
+        guard let service else {
+            return
+        }
+        var pages: [WKGrowthTask.Page] = []
+
+        let parameters = [ "action": "query",
+                           "generator": "growthtasks",
+                           "formatversion": "2",
+                           "format": "json",
+                           "ggttasktypes": "\(task.rawValue)",
+                           "ggtlimit": "10"
+        ]
+
+        guard let url = URL.mediaWikiAPIURL(project: project) else {
+            return
```

Comment:
You can call `completion(.failure(WKDataControllerError.failureCreatingRequestURL)` before this return.